src/NUnitFramework/framework/Attributes/PlatformAttribute.cs

```diff
@@ -56,12 +56,20 @@ public class PlatformAttribute : IncludeExcludeAttribute, IApplyToTest
         /// <param name="test">The test to modify</param>
         public void ApplyToTest(Test test)
         {
-            if (test.RunState != RunState.NotRunnable && 
-                test.RunState != RunState.Ignored && 
-                !platformHelper.IsPlatformSupported(this))
+            try
             {
-                test.RunState = RunState.Skipped;
-                test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                if (test.RunState != RunState.NotRunnable &&
+                    test.RunState != RunState.Ignored &&
+                    !platformHelper.IsPlatformSupported(this))
+                {
+                    test.RunState = RunState.Skipped;
+                    test.Properties.Add(PropertyNames.SkipReason, platformHelper.Reason);
+                }
+            }
+            catch (Exception ex)
+            {
+                test.RunState = RunState.NotRunnable;
+                test.Properties.Add(PropertyNames.SkipReason, ex.Message);
             }
```

Comment:
An `InvalidPlatformException` sounds good to me.

Earlier, we caught all exceptions in `PlatformHelper`, so in addition to an `InvalidPlatformException` I expect we will still want to have a general catch for all types of exceptions here.

I'll update this pull request with these changes as soon as I'm able.